src/main/java/org/terasology/module/inventory/ui/ContainerScreen.java

```diff
@@ -29,15 +34,29 @@ public EntityRef get() {
         });
         inventory.setCellOffset(10);
 
+        UILabel containerTitle = find("containerTitle", UILabel.class);
         containerInventory = find("container", InventoryGrid.class);
+
+        EntityRef characterEntity = localPlayer.getCharacterEntity();
+        CharacterComponent characterComponent = characterEntity.getComponent(CharacterComponent.class);
         containerInventory.bindTargetEntity(new ReadOnlyBinding<EntityRef>() {
             @Override
             public EntityRef get() {
-                EntityRef characterEntity = localPlayer.getCharacterEntity();
-                CharacterComponent characterComponent = characterEntity.getComponent(CharacterComponent.class);
                 return characterComponent.predictedInteractionTarget;
             }
         });
+        containerTitle.bindText(new ReadOnlyBinding<String>() {
+            @Override
+            public String get() {
+                Prefab parentPrefab = characterComponent.predictedInteractionTarget.getParentPrefab();
+                DisplayNameComponent displayName = parentPrefab.getComponent(DisplayNameComponent.class);
+                if (displayName != null) {
+                    return displayName.name;
+                } else {
+                    return parentPrefab.getName();
+                }
```

Comment:
Because you asked yesterday: This is how this would look with using `Optional`:

```suggestion
                return Optional.ofNullable(parentPrefab.getComponent(DisplayNameComponent.class))
                        .map(displayName -> displayName.name)
                        .orElse(parentPrefab.getName());
```

Since this is within a binding, I'm not sure the additional allocation of an `Optional` might not be ideal, and the current code might indeed be the better solution here.